apis/projectcontour/v1/httpproxy.go

```diff
@@ -864,6 +864,17 @@ type HeaderHashOptions struct {
 	HeaderName string `json:"headerName,omitempty"`
 }
 
+// QueryParameterHashOptions contains options to configure a query parameter based hash
+// policy, used in request attribute hash based load balancing.
+type QueryParameterHashOptions struct {
+	// QueryParameterName is the name of the HTTP request query parameter that will be used to
+	// calculate the hash key. If the query parameter specified is not present on a
+	// request, no hash will be produced.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	QueryParameterName string `json:"queryParameterName,omitempty"`
```

Comment:
maybe simplify to just `ParameterName` and json field `parameterName`? a little more succinct (in retrospect `HeaderName` in `HeaderHashOptions` could have just been `Name`)

Reply:
Done